### PR TITLE
Powert : Tests

### PR DESCRIPTION
**Pull request by aarshkshah1992**
For #811. 

@anorth The bulk of tests in the followup PR will be for the cron processing as that is the one missing deep tests.

### CODE REVIEW

File: actors/builtin/power/power_test.go [hunk 1]
```diff
@@ -72,6 +72,243 @@ func TestConstruction(t *testing.T) {
...
+		p3 := []byte("test")
+		miner2 := tutil.NewIDAddr(t, 501)
+		ac.enrollCronEvent(rt, miner2, e2, p3)
+		events = ac.getEnrolledCronTicks(rt, e2)
+		found, err = events.Get(0, &evt)
+		require.NoError(t, err)
+		require.True(t, found)
+		require.EqualValues(t, p3, evt.CallbackPayload)
+		require.EqualValues(t, miner2, evt.MinerAddr)
+	})
```
Comment (anorth) [inline actors/builtin/power/power_test.go:208]: Please test enrolling for an epoch that is before the current epoch

File: actors/builtin/power/power_test.go [hunk 2]
```diff
@@ -72,6 +72,243 @@ func TestConstruction(t *testing.T) {
...
+	zeroPledge := abi.NewTokenAmount(0)
+
+	t.Run("qaPower was below threshold before fault", func(t *testing.T) {
+		rt, ac := basicPowerSetup(t)
+		ac.createMinerBasic(rt, owner, owner, miner)
+		ac.updateClaimedPower(rt, miner, smallPowerUnit, smallPowerUnit)
+
+		st := getState(rt)
+		require.True(t, st.TotalRawBytePower.IsZero())
+		require.True(t, st.TotalQualityAdjPower.IsZero())
```
Comment (anorth) [inline actors/builtin/power/power_test.go:233]: Check `TotalBytesCommitted` and `TotalQABytesCommitted` here, and after the fault report.

File: actors/builtin/power/power_test.go [hunk 3]
```diff
@@ -72,6 +72,243 @@ func TestConstruction(t *testing.T) {
...
+	})
+
+	t.Run("qaPower was above threshold before fault", func(t *testing.T) {
+		rt, ac := basicPowerSetup(t)
+		ac.createMinerBasic(rt, owner, owner, miner)
+		ac.updateClaimedPower(rt, miner, powerUnit, powerUnit)
+
+		st := getState(rt)
+		require.EqualValues(t, powerUnit, st.TotalRawBytePower)
+		require.EqualValues(t, powerUnit, st.TotalQualityAdjPower)
```
Comment (anorth) [inline actors/builtin/power/power_test.go:251]: Similarly show the values for BytesCommitted here

File: actors/builtin/power/power_test.go [hunk 4]
```diff
@@ -72,6 +72,243 @@ func TestConstruction(t *testing.T) {
...
+
+		delta := abi.NewTokenAmount(100)
+		ac.updatePledgeTotal(rt, miner, delta)
+
+		slash := abi.NewTokenAmount(50)
+		ac.onConsensusFault(rt, miner, &slash)
+
+		st := getState(rt)
+		require.EqualValues(t, big.Sub(delta, slash), st.TotalPledgeCollateral)
+	})
```
Comment (anorth) [inline actors/builtin/power/power_test.go:275]: I suggest folding this into the tests above.

File: actors/builtin/power/power_test.go [hunk 5]
```diff
@@ -581,6 +846,28 @@ func (h *spActorHarness) getClaim(rt *mock.Runtime, a addr.Address) *power.Claim
 	return &out
 }
 
+func (h *spActorHarness) getEnrolledCronTicks(rt *mock.Runtime, epoch abi.ChainEpoch) *adt.Array {
```
Comment (anorth) [inline actors/builtin/power/power_test.go:849]: Consider extracting the `Array` to a Go slice for easier inspection in the test methods

Comment (aarshkshah1992) [reply actors/builtin/power/power_test.go]: This is done.

File: actors/builtin/power/power_test.go (hunk 1 above)
Comment (aarshkshah1992) [reply actors/builtin/power/power_test.go:198]: This is done.

File: actors/builtin/power/power_test.go (hunk 3 above)
Comment (aarshkshah1992) [reply actors/builtin/power/power_test.go:274]: This is done.

File: actors/builtin/power/power_test.go (hunk 2 above)
Comment (aarshkshah1992) [reply actors/builtin/power/power_test.go:252]: This is done.

File: actors/builtin/power/power_test.go (hunk 4 above)
Comment (aarshkshah1992) [reply actors/builtin/power/power_test.go:292]: This is done.

**Pull request merged**